sdk/tests/deqp/functional/gles3/es3fShaderStructTests.js

```diff
@@ -0,0 +1,1951 @@
+		/** @type {string} */ var defaultVertSrc =
+			'#version 300 es\n' +
+			'in highp vec4 a_position;\n' +
+			'in highp vec4 a_coords;\n' +
+			'out mediump vec4 v_coords;\n\n' +
+			'void main (void)\n' +
+			'{\n' +
+			'	v_coords = a_coords;\n' +
+			'	gl_Position = a_position;\n' +
+			'}\n';
+		/** @type {string} */ var defaultFragSrc =
+			'#version 300 es\n' +
+			'in mediump vec4 v_color;\n' +
+			'layout(location = 0) out mediump vec4 o_color;\n\n' +
+			'void main (void)\n' +
+			'{\n' +
+			'	o_color = v_color;\n' +
+			'}\n';
+
+		// Fill in specialization parameters.
+		var spParams = {};
+		if (isVertexCase) {
+			spParams["HEADER"] =
+				"#version 300 es\n" +
+				"in highp vec4 a_position;\n" +
+				"in highp vec4 a_coords;\n" +
+				"out mediump vec4 v_color;";
+			spParams["COORDS"]		= "a_coords";
+			spParams["DST"]			= "v_color";
+			spParams["ASSIGN_POS"]	= "gl_Position = a_position;";
+		}
+		else {
+			spParams["HEADER"]	=
+				"#version 300 es\n" +
+				"in mediump vec4 v_coords;\n" +
+				"layout(location = 0) out mediump vec4 o_color;";
+			spParams["COORDS"]			= "v_coords";
+			spParams["DST"]				= "o_color";
+			spParams["ASSIGN_POS"]		= "";
+		}
+
+		if (isVertexCase)
+			return new es3fShaderStructTests.ShaderStructCase(name, description, isVertexCase, usesTextures, evalFunc, setupUniforms, tcuStringTemplate.specialize(shaderSrc, spParams), defaultFragSrc);
+		else
+			return new es3fShaderStructTests.ShaderStructCase(name, description, isVertexCase, usesTextures, evalFunc, setupUniforms, defaultVertSrc, tcuStringTemplate.specialize(shaderSrc, spParams));
+	};
+
+	/**
+	 * @constructor
+	 * @extends {tcuTestCase.DeqpTest}
+	 */
+	es3fShaderStructTests.LocalStructTests = function() {
+		tcuTestCase.DeqpTest.call(this, 'local', 'Local structs');
+		this.makeExecutable();
+	};
+
+	es3fShaderStructTests.LocalStructTests.prototype = Object.create(tcuTestCase.DeqpTest.prototype);
+	es3fShaderStructTests.LocalStructTests.prototype.constructor = es3fShaderStructTests.LocalStructTests;
+
+	es3fShaderStructTests.LocalStructTests.prototype.init = function() {
+		var currentCtx = this;
+		function LocalStructCase(name, description, shaderSource, evalFunction) {
+			currentCtx.addChild(es3fShaderStructTests.ShaderStructCase.createStructCase(name + "_vertex", description, true, false, evalFunction, null, shaderSource));
+			currentCtx.addChild(es3fShaderStructTests.ShaderStructCase.createStructCase(name + "_fragment", description, false, false, evalFunction, null, shaderSource));
+		};
+
+		LocalStructCase('basic', 'Basic struct usage',
+			'${HEADER}\n' +
+			'uniform int ui_one;\n' +
+			'\n' +
+			'struct S {\n' +
+			'	mediump float	a;\n' +
+			'	mediump vec3	b;\n' +
+			'	int				c;\n' +
+			'};\n' +
+			'\n' +
+			'void main (void)\n' +
+			'{\n' +
+			'	S s = S(${COORDS}.x, vec3(0.0), ui_one);\n' +
+			'	s.b = ${COORDS}.yzw;\n' +
+			'	${DST} = vec4(s.a, s.b.x, s.b.y, s.c);\n' +
+			'	${ASSIGN_POS}\n' +
+			'}\n',
```

Comment:
I hope these shaders were a simple copy-paste and reformat from es3fShaderStructTests.cpp! :)